hivemind/optim/ddp.py

```diff
@@ -0,0 +1,178 @@
+import time
+from typing import Callable, Optional, Union
+
+import torch
+from torch.distributed.distributed_c10d import _get_default_group, _get_default_store
+
+from hivemind.dht import DHT
+from hivemind.optim.grad_scaler import GradScaler
+from hivemind.optim.optimizer import Optimizer
+from hivemind.optim.state_averager import OptimizerFactory, Parameters, ParamGroups, TorchOptimizer, TrainingStateAverager
+from hivemind.utils import get_logger
+
+logger = get_logger(__name__)
+
+
+class DDPOptimizer(Optimizer):
```

Comment:
**Note to self:** A better way to do it is:

- Don't inherit `hivemind.Optimizer`
- Make `_create_optimizer()` method and forward `__init__`'s kwargs there
- Make `opt` property
- _Maybe_ create `__getattr__` that can forward attrs to `opt`